ReactNativeClient/lib/joplin-database.js

```diff
@@ -735,6 +735,87 @@ class JoplinDatabase extends Database {
 				);
 			}
 
+			if (targetVersion == 31) {
+				queries.push('DROP TRIGGER notes_fts_before_update');
+				queries.push('DROP TRIGGER notes_fts_before_delete');
+				queries.push('DROP TRIGGER notes_after_update');
+				queries.push('DROP TRIGGER notes_after_insert');
+
+				queries.push('DROP INDEX notes_normalized_id');
+				queries.push('DROP TABLE notes_normalized');
+				queries.push('DROP TABLE notes_fts');
+
```

Comment:
If tagetVersion 15 and 18 fail, we're not throwing an error. We just won't use FTS. 

```js
 try {
     await this.transactionExecBatch(queries);
 } catch (error) {
     if (targetVersion === 15 || targetVersion === 18) {
     this.logger().warn('Could not upgrade to database v15 or v18 - FTS feature will not be used', error);
 } else {
     throw error;
     }
 }
```

But if these versions fail, we won't have the triggers created by them for keeping the FTS table in sync.

So the line
```js
queries.push('DROP TRIGGER notes_fts_before_update');
```
 in targetVersion 31 will error because that trigger won't exist. Is it better to warn the user instead of throwing an error, like we're doing for v15 and v18?

Reply:
Yes I guess the logic should be the same.